nvme.c

```diff
@@ -3331,8 +3331,7 @@ static int list_subsys(int argc, char **argv, struct command *cmd,
 
 	err = nvme_scan_topology(r, filter, (void *)devname);
 	if (err) {
-		if (errno != ENOENT)
-			nvme_show_error("Failed to scan topology: %s", nvme_strerror(errno));
+		nvme_show_error("Failed to scan topology: %s", nvme_strerror(errno));
```

Comment:
Personally feel like `nvme list-subsys` and `nvme list` should succeed with empty output if the modules are unloaded. But if others disagree, I am fine with keeping it this way.

Reply:
Really? I can understand making an exception for ```disconnect-all```, but do we want this for  ```list``` and ```list-subsys``` too? I'd prefer not...

Reply:
Yes, that's why I suggested doing this in the implementation of `nvme_scan_topology()` so we wouldn't have to modify all the call sites. If the module isn't loaded, then the topology should be empty.
But as I said, I don't feel strongly about `nvme list`/`list-subsys`. I think it would be more consistent to print an empty list, but we're not relying on it in our workflow.

Reply:
nvme-cli is not the only user of the library. I generally try to avoid filtering out errors in the middle of a call stack unless we can resolve them. But in this case some users want to see the error others not. See this very discussion.